src/test/java/hudson/plugins/active_directory/docker/TheFlintstonesIT.java

```diff
@@ -60,6 +74,37 @@ public class TheFlintstonesIT {
     @Rule
     public LoggerRule l = new LoggerRule();
 
+    @Before
+    public void overrideDNSServers() throws UnknownHostException {
+        // we need to pint the JVMs DNS resolver at the AD (samba) server
+        // for AD to work correctly it needs to be able to resolve hosts and do SRV lookups on the domain
+
+        // whilst the `getHost()` is supposed to return an IPAddress in some cases it will return "localhost"
+        // we need a resolved address to configure the resolver do a lookup before we change the DNS.
+        InetAddress hostInetAddr = InetAddress.getByName(docker.getHost());
+        String hostIP = hostInetAddr.getHostAddress();
+
+        // but additionally we need to use the locally bound ports for the catalog and not what AD returns for name resolution
+        Map<String, String> hostAliases = new LinkedHashMap<>();
+        hostAliases.put("dc1.samdom.example.com", hostIP);
+        // this adds the A entry for the PDC, but will leave the discovery of this to the SRV lookup
+
+        HostResolutionRequestInterceptor.INSTANCE.install(
+                new MappedHostResolver(hostAliases),
+                new DNSClientHostResolver(hostIP, 553),
+                DefaultHostResolver.INSTANCE);
+
+        // we also need to set the JNDI default
+        // see hudson.plugins.active_directory.ActiveDirectoryDomain.createDNSLookupContext()
+        System.setProperty(DNSUtils.OVERRIDE_DNS_PROPERTY, "dns://"+hostIP+":553");
```

Comment:
Note: URL construction as string may not work on IPv6.

Reply:
good catch.

Reply:
e7b692d493e2b9dedd13249261afb363e5f4b8b1